docs/source-pytorch/advanced/model_parallel/fsdp.rst

```diff
@@ -2,138 +2,439 @@
 
 .. _fully-sharded-training:
 
+###################################################
+Train models with billions of parameters using FSDP
+###################################################
+
+Use Fully Shared Data Parallel (FSDP) to train large models with billions or trillions of parameters efficiently on multiple GPUs and across multiple machines.
+
+.. note:: This is an experimental feature.
+
+
+Today, large models with billions of parameters are trained with many GPUs across several machines in parallel.
+Even a single A100 GPU with 80 GB of VRAM (the biggest today) is not enough to train just a 30B parameter model (even with batch size 1 and 16-bit precision).
+The memory consumption for training is generally made up of
+
+1. the model parameters,
+2. the optimizer states (e.g., Adam has two additional exponential averages per parameter),
```

Comment:
super nit: put optimizer states last, since people generally think of fwd>bwd>grad